From 07dbae1f59592d4da0b6a248ba24506d23f1dcd9 Mon Sep 17 00:00:00 2001
From: Christian Breunig <christian@breunig.cc>
Date: Thu, 8 Feb 2024 21:53:35 +0100
Subject: [PATCH] ospfd: can not delete "segment-routing node-msd" when SR if
 off

This fixes the initial implementation of commit 7743f2f8c00 ("OSPFd: Update
Segment Routing PR following review") where it wsa not possible to remove
the "segment-routing node-msd" CLI nodes via vtysh once segment-routing got
disabled.

Closes #14910

Signed-off-by: Christian Breunig <christian@breunig.cc>
(cherry picked from commit ba45aa62e7bcc4ea7b02773614f03efb6f7c595b)
---
 ospfd/ospf_sr.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ospfd/ospf_sr.c b/ospfd/ospf_sr.c
index 467cb0504..e26fe6f53 100644
--- a/ospfd/ospf_sr.c
+++ b/ospfd/ospf_sr.c
@@ -580,6 +580,7 @@ static void ospf_sr_stop(void)
 	hash_clean(OspfSR.neighbors, (void *)sr_node_del);
 	OspfSR.self = NULL;
 	OspfSR.status = SR_OFF;
+	OspfSR.msd = 0;
 }
 
 /*
-- 
2.39.2

